framework_tvb/tvb/core/services/burst_service.py

```diff
@@ -235,3 +239,73 @@ def prepare_name(burst, project_id):
             default_simulation_name = burst.name
 
         return default_simulation_name, simulation_number
+
+    def prepare_indexes_for_simulation_results(self, operation, result_filenames, burst):
+        indexes = list()
+        self.logger.debug("Preparing indexes for simulation results in operation {}...".format(operation.id))
+        for filename in result_filenames:
+            index = h5.index_for_h5_file(filename)()
+            # TODO: don't load full TS in memory and make this read nicer
+            datatype, ga = h5.load_with_references(filename)
+            index.fill_from_has_traits(datatype)
+            index.fill_from_generic_attributes(ga)
+            index.fk_parent_burst = burst.id
+            index.fk_from_operation = operation.id
+            if operation.fk_operation_group:
+                datatype_group = dao.get_datatypegroup_by_op_group_id(operation.fk_operation_group)
+                self.logger.debug(
+                    "Found DatatypeGroup with id {} for operation {}".format(datatype_group.id, operation.id))
+                index.fk_datatype_group = datatype_group.id
+            self.logger.debug(
+                "Prepared index {} for file {} in operation {}".format(index.summary_info, filename, operation.id))
+            indexes.append(index)
+        self.logger.debug("Prepared {} indexes for results in operation {}...".format(len(indexes), operation.id))
+        return indexes
+
+    def prepare_index_for_metric_result(self, operation, result_filename, burst):
+        self.logger.debug("Preparing index for metric result in operation {}...".format(operation.id))
+        index = DatatypeMeasureIndex()
+        with DatatypeMeasureH5(result_filename) as dti_h5:
+            index.metrics = json.dumps(dti_h5.metrics.load())
+            index.fk_source_gid = dti_h5.analyzed_datatype.load().hex
+        index.fk_from_operation = operation.id
+        index.fk_parent_burst = burst.id
+        datatype_group = dao.get_datatypegroup_by_op_group_id(operation.fk_operation_group)
+        self.logger.debug("Found DatatypeGroup with id {} for operation {}".format(datatype_group.id, operation.id))
+        index.fk_datatype_group = datatype_group.id
+        self.logger.debug("Prepared index {} for results in operation {}...".format(index.summary_info, operation.id))
+        return index
+
+    def update_finished_burst_status(self, burst_config):
+        burst_service = BurstService()
```

Comment:
Isn't this self?

Reply:
It is, missed it when moving the method.